### PR TITLE
fix(unlock-app): UI improvements to handle connections better

### PR DESCRIPTION
**Pull request by julien51**
# Description

A few small changes to handle provider connections better.

See comments.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/components/interface/checkout/Connected.tsx [hunk 1]
```diff
@@ -189,31 +189,39 @@ export function Connected({
   })
   const { signIn, signOut, isSignedIn } = useSIWE()
   const [isDisconnecting, setIsDisconnecting] = useState(false)
-  const autoconnect = state.context?.paywallConfig?.autoconnect
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/Connected.tsx:192]: We don't use `autoconnect` anymore...

File: unlock-app/src/hooks/useAutoLogin.ts [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
   const { authenticateWithProvider } = useAuthenticate()
+  const { signOut } = useSIWE()
 
   const getAutoLoginData = useCallback(() => {
     const storedProvider: WalletProvider | null = getStorage(
       'provider'
     ) as WalletProvider
-
-    const canAutoLogin = storedProvider !== null
+    const canAutoLogin =
```
Comment (julien51) [inline unlock-app/src/hooks/useAutoLogin.ts:16]: Delegated providers can't autologin...

File: unlock-app/src/components/interface/checkout/index.tsx [hunk 3]
```diff
@@ -46,13 +46,19 @@ export function CheckoutPage() {
...
   useEffect(() => {
-    authenticateWithProvider('DELEGATED_PROVIDER', injectedProvider)
-  }, [authenticateWithProvider, injectedProvider])
+    if (communication.providerAdapter) {
+      authenticateWithProvider(
+        'DELEGATED_PROVIDER',
+        communication.providerAdapter
+      )
+    }
+  }, [authenticateWithProvider, communication.providerAdapter])
```
Comment (julien51) [inline unlock-app/src/components/interface/checkout/index.tsx:57]: We only do the above for `communication.providerAdapter`... not for `selectProvider(config)`

File: unlock-app/src/hooks/useAutoLogin.ts [hunk 4]
```diff
@@ -33,10 +35,14 @@ export function useAutoLogin() {
       } catch (error: any) {
         console.error('Autologin failed.')
         console.error(error)
+        await signOut()
       }
     }
+    if (!canAutoLogin && storedProvider === 'DELEGATED_PROVIDER') {
```
Comment (julien51) [inline unlock-app/src/hooks/useAutoLogin.ts:41]: Remove the token if autologin fails for `DELEGATED_PROVIDER` ( we keep it for unlock accounts!)

File: unlock-app/src/components/interface/checkout/index.tsx [hunk 5]
```diff
@@ -46,13 +46,19 @@ export function CheckoutPage() {
...
+
+  // TODO: do we need to pass it down?
+  const injectedProvider =
+    communication.providerAdapter || selectProvider(config)
```
Comment (searchableguy) [inline unlock-app/src/components/interface/checkout/index.tsx:61]: Yes

**Review by searchableguy — APPROVED** [review]

**Pull request merged**